### ISSUE TITLE
Launching ur10e with humble in simulation mode

### ISSUE THREAD

**Issue by skaeringur97** [UniversalRobots/Universal_Robots_ROS2_Driver#503]
I was trying to get a simple simulation running with the universal robot ur10e, but it took many hours to get working due to misleading/wrong documentation. 

Firstly, to start the driver, I ran 
```
ros2 launch ur_robot_driver ur_control.launch.py ur_type:=ur10e robot_ip:=yyy.yyy.yyy.yyy use_fake_hardware:=true launch_rviz:=true
```
but when I checked active controllers, I see that scaled_joint_trajectory_controller is active which doesn't work in simulation:

```
ros2 control list_controllers
forward_position_controller[position_controllers/JointGroupPositionController] inactive  
io_and_status_controller[ur_controllers/GPIOController] active    
scaled_joint_trajectory_controller[ur_controllers/ScaledJointTrajectoryController] active    
speed_scaling_state_broadcaster[ur_controllers/SpeedScalingStateBroadcaster] active    
joint_state_broadcaster[joint_state_broadcaster/JointStateBroadcaster] active    
force_torque_sensor_broadcaster[force_torque_sensor_broadcaster/ForceTorqueSensorBroadcaster] active 
```

The use_fake_hardware argument is supposed to change the default controller from scaled_joint_trajectory_controller to joint_trajectory_controller, but for some reason it doesn't work.

My workaround solution for this is to add 'initial_joint_controller:=joint_trajectory_controller' to the arguments. That works to activate the correct controller for simulation:
```
ros2 control list_controllers
joint_trajectory_controller[joint_trajectory_controller/JointTrajectoryController] active    
joint_state_broadcaster[joint_state_broadcaster/JointStateBroadcaster] active    
force_torque_sensor_broadcaster[force_torque_sensor_broadcaster/ForceTorqueSensorBroadcaster] active    
speed_scaling_state_broadcaster[ur_controllers/SpeedScalingStateBroadcaster] active    
io_and_status_controller[ur_controllers/GPIOController] active    
forward_position_controller[position_controllers/JointGroupPositionController] inactive
```

Next, the moveit node needs be executed. The only documentation about that one is to run this line:
```
ros2 launch ur_moveit_config ur_moveit.launch.py ur_type:=ur10e launch_rviz:=true
```

However, that doesn't work for simulation. I checked the source code and saw that it also includes use_fake_hardware. So, I thought I could just add that to the arguments and that would allow me to run it in simulation. However, that only gave me a new error:
```
Action client not connected to action server: scaled_joint_trajectory_controller/follow_joint_trajectory
```
After a lot of debugging, I finally noticed another argument use_sim_time that needs to be manually set to true, when using simulation. Finally, when adding that to my arguments, I could plan and execute plans.

So to summarize. The switch of controllers in ur_robot_driver, when using fake hardware, needs to be fixed. Also, better documentation for ur_moveit_config is needed. I would also think that the 'use_fake_hardware' should activate 'use_sim_time' automatically, unless there is a situation that I'm missing where you need them to be separate.

**Comment by fmauch** [comment]
This sounds like it should be resolved by #480. If you build from the current main branch, does that still persist?

**Comment by skaeringur97** [comment]
Yes, I tried both building from source and binary build and the problem remains in both.

**Comment by fmauch** [comment]
Checking this again, I find the error. As you have correctly said, currently it is needed to specify  `initial_joint_controller:=joint_trajectory_controller` when using fake_hardware. However, MoveIt! does need to get the information about fake_hardware, as well, in order to connect to the correct controller (The change mentioned in #480).

So, the following two launches should get you running:

```
ros2 launch ur_robot_driver ur_control.launch.py ur_type:=ur10e robot_ip:=yyy.yyy.yyy.yyy use_fake_hardware:=true launch_rviz:=true initial_joint_controller:=joint_trajectory_controller
```
```
ros2 launch ur_moveit_config ur_moveit.launch.py ur_type:=ur10e launch_rviz:=true use_fake_hardware:=true
```

You are correct, the docs do not reflect that correctly at the moment. That's definitely something for #494, although this will change in the near future again, as soon as https://github.com/ros-controls/ros2_control/pull/822 gets merged.

**Comment by skaeringur97** [comment]
>ros2 launch ur_moveit_config ur_moveit.launch.py ur_type:=ur10e launch_rviz:=true use_fake_hardware:=true

If I'm not mistaken, this line needs to be:
```
ros2 launch ur_moveit_config ur_moveit.launch.py ur_type:=ur10e launch_rviz:=true use_fake_hardware:=true use_sim_time:=true
```

**Comment by fmauch** [comment]
no, use_sime_time is not really connected to that, as this is not a real simulation such as gazebo, but (as the name suggests) a fake hardware. It acts like a real hardware interface at the communication level, but it is not a full robot simulation.

**Comment by skaeringur97** [comment]
hmm, weird then that it didn't work for me until I added use_sim_time:=true

**Comment by skaeringur97** [comment]
I noticed that, if I copy the ur_moveit.launch.py exactly as it is [on github](https://github.com/UniversalRobots/Universal_Robots_ROS2_Driver/blob/main/ur_moveit_config/launch/ur_moveit.launch.py) to my own package and run it from there, then I don't need to have use_sim_time:=true for it to execute a plan. Running it from my binary install still fails in execution without use_sim_time:=true

**Comment by fmauch** [comment]
The necessary changes will come with the [next sync](https://discourse.ros.org/t/preparing-for-humble-sync-and-patch-release-2022-11-21/28234). I don't quite understand why the `use_sim_time` parameter should make things work without those changes, but since it is working with the recent version, I'm going ahead and close this.
